### PR TITLE
[CAP-108] [BE] - Implement attendant create application

### PR DESCRIPTION
**Pull request by CaoSon1909**
- 19/02/2022: work in progress

### CODE REVIEW

File: src/main/java/org/capstone/job_fair/controllers/payload/requests/account/cv/CreateApplicationRequest.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import javax.validation.constraints.NotEmpty;
+
+@Data
+@AllArgsConstructor
+@ToString
+@NoArgsConstructor
+public class CreateApplicationRequest {
+    @XSSConstraint
+    @NotEmpty
+    private String summary;
```
Comment (TienTruong2000) [inline src/main/java/org/capstone/job_fair/controllers/payload/requests/account/cv/CreateApplicationRequest.java:18]: No constraint????

File: src/main/java/org/capstone/job_fair/services/impl/attendant/cv/ApplicationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import org.capstone.job_fair.repositories.company.job.RegistrationJobPositionRepository;
+import org.capstone.job_fair.services.interfaces.account.AccountService;
+import org.capstone.job_fair.services.interfaces.attendant.ApplicationService;
+import org.capstone.job_fair.utils.MessageUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service
```
Comment (TienTruong2000) [inline src/main/java/org/capstone/job_fair/services/impl/attendant/cv/ApplicationServiceImpl.java:23]: Where is @Transactional

File: src/main/java/org/capstone/job_fair/controllers/attendant/cv/ApplicationController.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            //
+            dto.setSummary(request.getSummary());
+            dto.setCreateDate(new Date().getTime());
+            dto.setStatus(Application.DRAFT);
+            dto.setAttendantDTO(attendantDTO);
+            dto.setRegistrationJobPositionDTO(regisDTO);
+            ApplicationDTO result = applicationService.createNewApplication(dto);
+            return ResponseEntity.status(HttpStatus.CREATED).body(result);
+        }
+        catch (NoSuchElementException | IllegalArgumentException ex) {
```
Comment (TienTruong2000) [inline src/main/java/org/capstone/job_fair/controllers/attendant/cv/ApplicationController.java:64]: Remove catch please

File: src/main/java/org/capstone/job_fair/services/impl/attendant/cv/ApplicationServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        registrationJobPositionEntity.setId(dto.getRegistrationJobPositionDTO().getId());
+        entity.setRegistrationJobPosition(registrationJobPositionEntity);
+        entity = cvRepository.save(entity);
+
+        //
+        ApplicationDTO resultDTO = new ApplicationDTO();
+        resultDTO.setId(entity.getId());
+        resultDTO.setSummary(entity.getSummary());
+        resultDTO.setCreateDate(entity.getCreateDate());
+        resultDTO.setStatus(entity.getStatus());
```
Comment (TienTruong2000) [inline src/main/java/org/capstone/job_fair/services/impl/attendant/cv/ApplicationServiceImpl.java:73]: comment clearer please

**Review by TienTruong2000 — CHANGES_REQUESTED** [review]
Change please

**Review by BuiMinhKhoi — APPROVED** [review]

**Review by BuiMinhKhoi — APPROVED** [review]

**Review by BuiMinhKhoi — APPROVED** [review]

**Review by TienTruong2000 — APPROVED** [review]
LGTM

**Pull request merged**